packages/react-ui-components/src/DropDown/header.tsx

```diff
@@ -56,6 +56,15 @@ export interface ShallowDropDownHeaderProps {
 
     readonly toggleDropDown: () => void;
 
+    /**
+     * This prop gets access to the onClick function for reference editor. It changes the url.
+     */
+    readonly onReferenceClick?: (args: string) => void;
```

Comment:
This shouldn't be called `reference` as mentioned before. This is a presentational component and shouldn't have deeper knowledge about its usage. So `onItemClick` or something similar should be better.